resolver/resolver_test.go

```diff
@@ -309,127 +84,168 @@ func TestNonMesosHandler(t *testing.T) {
 		if err != nil {
 			return nil, err
 		}
-		msg := &dns.Msg{
-			Answer: []dns.RR{rr1, rr2},
-		}
+		msg := &dns.Msg{Answer: []dns.RR{rr1, rr2}}
 		msg.SetReply(r)
 		return msg, nil
 	}
-	if err != nil {
-		t.Error(err)
-	}
-
-	dns.HandleFunc(".", res.HandleNonMesos)
-	safeStartDNSResolver(t, res)
-
-	// test A records
-	msg, err = fakeQuery("google.com", dns.TypeA, "udp", port)
-	if err != nil {
-		t.Error(err)
-	}
 
-	if len(msg) < 1 {
-		t.Errorf("not serving up A records, expected 2 records instead of %d: %+v", len(msg), msg)
+	for i, tt := range []struct {
+		handle dns.HandlerFunc
+		query  dns.Msg
+		assert func(*dns.Msg) []error
+	}{
+		{
+			res.HandleMesos,
+			query("chronos.marathon.mesos.", dns.TypeA),
+			message(
+				header(0, 0, true, dns.RcodeSuccess),
```

Comment:
They are typed. However, just like any other API, it takes time learning. Since the first two arguments are constant as of now, we could simply drop them.